ESP8266Interface.cpp

```diff
@@ -64,20 +65,35 @@ int ESP8266Interface::connect(const char *ssid, const char *pass, nsapi_security
 
 int ESP8266Interface::connect()
 {
+    nsapi_error_t fw_status = get_firmware_ok();
+    size_t ssid_length = strlen(ap_ssid);
+    size_t pass_length = strlen(ap_pass);
+
+    _esp.setTimeout(ESP8266_MISC_TIMEOUT);
+
+    if(_esp.isConnected()) {
+        return NSAPI_ERROR_ALREADY;
+    }
+
     _esp.setTimeout(ESP8266_CONNECT_TIMEOUT);
-    
     if (!_esp.reset()) {
         return NSAPI_ERROR_DEVICE_ERROR;
-    }   
- 
-    _esp.setTimeout(ESP8266_MISC_TIMEOUT);
-    
-    if (_esp.get_firmware_version() != ESP8266_VERSION) {
-        debug("ESP8266: ERROR: Firmware incompatible with this driver.\
-               \r\nUpdate to v%d - https://developer.mbed.org/teams/ESP8266/wiki/Firmware-Update\r\n",ESP8266_VERSION); 
-        return NSAPI_ERROR_DEVICE_ERROR;
     }
-    
+
+    if (ssid_length == 0) {
+        return NSAPI_ERROR_NO_SSID;
+    }
+
+    if (ap_sec != NSAPI_SECURITY_NONE) {
+        if (pass_length < ESP8266_PASSPHRASE_MIN_LENGTH) {
+            return NSAPI_ERROR_PARAMETER;
+        }
+    }
+
+    if (fw_status != NSAPI_ERROR_OK) {
```

Comment:
I would prefer to initialize the parameter as late as possible. Or maybe this fw_status should be checked straight after the isConnected check?

Reply:
Or maybe even before icConnected? Who knows how request for IP address might change - I doubt it but possible.